src/main/scala/is/hail/utils/package.scala

```diff
@@ -483,4 +483,19 @@ package object utils extends Logging
     finally
       resourceStream.close()
   }
+
+  def roundWithConstantSum(a: Array[Double]): Array[Int] = {
+    val withFloors = a.zipWithIndex.map { case (d, i) => (i, d, math.floor(d)) }
+    val totalFractional = (withFloors.map { case (i, orig, floor) => orig - floor }.sum + 0.5).toInt
```

Comment:
The code I wrote was only for N=3. Otherwise, I needed to write a generic sort. The function is on line 134 here: https://github.com/jigold/hail/blob/1756ff933905a3f4d290f56f1c6dac3e0fc967d4/src/main/scala/is/hail/io/bgen/BgenWriter.scala